tests/runtime/test_batch.py

```
@@ -0,0 +1,533 @@
+
+    def test_both_reset_after_exit(self):
+        client = MockClient()
+        with BatchJobSession(client=client):
+            pass
+        assert get_active_batch() is None
+        assert get_active_batch_session() is None
+
+    def test_context_reset_even_on_exception(self):
+        client = MockClient()
+        with pytest.raises(RuntimeError):
+            with BatchJobSession(client=client):
```

Ruff (0.15.7)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)